test/Date/DateParse.js

```diff
@@ -174,3 +174,57 @@ WScript.Echo(Date.parse("2011/11-08 19:48:43"));
 
 // Multiple 'T's aren't allowed
 WScript.Echo(Date.parse("2011-11-08TT19:48:43"));
+
+// Verifying non-standard delimiters with milliseconds
+
+// Missing digits after . or :
+WScript.Echo(Date.parse("2011-11-08 19:48:43."));
+WScript.Echo(Date.parse("2011-11-08 19:48:43:"));
+
+// milliseconds, 1 digit only, treat it as hundreds
+WScript.Echo(Date.parse("2011-11-08 19:48:43:1"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.1"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.1a"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.0"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.0a"));
+
+// milliseconds, 2 digit only, treat it as  tens
+WScript.Echo(Date.parse("2011-11-08 19:48:43.12"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.12a"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.01"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.01a"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.00"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.00a"));
+
+// milliseconds, canonical 3 digit, per EcmaScript spec
+WScript.Echo(Date.parse("2011-11-08 19:48:43.123"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.123a"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.001"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.001a"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.000"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.000a"));
+
+// milliseconds, with more than 3 digits, ignore all digits after the third
+WScript.Echo(Date.parse("2011-11-08 19:48:43.1234"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.1234a"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.0011"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.0011a"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.0001"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.0001a"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.0000"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.0000a"));
+
+WScript.Echo(Date.parse("2011-11-08 19:48:43.12345"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.12345a"))
+WScript.Echo(Date.parse("2011-11-08 19:48:43.00111"));;
+WScript.Echo(Date.parse("2011-11-08 19:48:43.00111a"));;
+WScript.Echo(Date.parse("2011-11-08 19:48:43.00001"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.00001a"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.00000"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.00000a"));
+
+WScript.Echo(Date.parse("2011-11-08 19:48:43.00000000000001"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.00000000000001a"));
+
+WScript.Echo(Date.parse("2011-11-08 19:48:43.12340000000001"));
+WScript.Echo(Date.parse("2011-11-08 19:48:43.12340000000001a"));
```

Comment:
nit: newline at end of file
You probably need to update the test baseline as well.